pkg/filters/proxy/pool.go

```diff
@@ -635,6 +636,8 @@ func (sp *ServerPool) buildResponse(spCtx *serverPoolContext) (err error) {
 		resp = r
 	}
 
+	setCookies(resp, spCtx.respCookies)
```

Comment:
I would prefer to add a new method to the `LoadBalancer` interface for this task.
maybe the method name could be `BindToServer`, the `StickySessionsLoadBalancer` set the cookies, while other load balancers do nothing.